File: typescript/src/services/search-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { SearchDto, SearchShareDto, SearchSuggestionDto } from '../models/index.js';


export class SearchApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async pageSearch(request: SearchDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`search`, request);
  }

  async searchSuggestions(request: SearchSuggestionDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`search/suggest`, request);
  }

  async searchShare(request: SearchShareDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`search/share-search`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/SearchApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class SearchApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> PageSearchAsync(SearchDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("search", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> SearchSuggestionsAsync(SearchSuggestionDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("search/suggest", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> SearchShareAsync(SearchShareDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("search/share-search", request, cancellationToken).ConfigureAwait(false);
    }

}
